src/collectors/postgres/postgres.py

```diff
@@ -79,53 +103,81 @@ def collect(self):
             self.log.error('Unable to import module psycopg2')
             return {}
 
-        # Get list of databases
-        dbs = self._get_db_names()
-        if len(dbs) == 0:
-            self.log.error("I have 0 databases!")
-            return {}
-
-        if self.config['metrics']:
-            metrics = self.config['metrics']
-        elif str_to_bool(self.config['extended']):
-            metrics = registry['extended']
-            if str_to_bool(self.config['has_admin']) \
-                    and 'WalSegmentStats' not in metrics:
-                metrics.append('WalSegmentStats')
+        instances = self.config.get('instances')
+
+        # HACK: setting default with subcategory messes up merging of configs,
+        # so we only set the default if one wasn't provided.
+        if not instances:
```

Comment:
Won't this code never run since you defined a default for instances that is not None?